storage/tick.go

```diff
@@ -125,36 +143,39 @@ func (mgr *tickManager) Tick(softDeadline time.Duration, forceType forceType) er
 
 	// Begin ticking
 	start := mgr.nowFn()
-	sizes := make([]int64, 0, len(namespaces))
-	totalSize := int64(0)
 	for _, n := range namespaces {
-		size := n.NumSeries()
-		sizes = append(sizes, size)
-		totalSize += size
+		n.Tick(mgr.c)
 	}
-	for i, n := range namespaces {
-		deadline := float64(softDeadline) * (float64(sizes[i]) / float64(totalSize))
-		n.Tick(mgr.c, time.Duration(deadline))
+
+	// NB(r): Always sleep for some constant period since ticking
+	// is variable with num series. With a really small amount of series
+	// the per shard amount of constant sleeping is only:
+	// = num shards * min sleeps per shard (32 constant) * sleep per series (100 microseconds default)
+	// This number can be quite small if configuring to have small amount of
+	// shards (say 10 shards):
+	// = 10 num shards * 32 min sleeps * 100 microseconds default sleep per series
+	// = 10*32*0.1 milliseconds
+	// = 32ms
+	// Because of this we always sleep at least some fixed constant amount.
```

Comment:
Whats the benefit of having this minimum sleep in-between ticks versus just "always be ticking"? Presumably if you have a lot of series, you're always gonna be ticking anyways (say it takes 15 minutes to tick and then you wait one minute inbetween, you're ticking the vast majority of the time). Would it be better to have the performance be constant vs spiky?

Reply:
Yeah this is to catch the super small embedded case, e.g. M3DB as a local time series stores on your application host.